libControls/ToolTip.cpp

```diff
@@ -42,7 +43,21 @@ void ToolTip::add(Control& c, const std::string& str)
 
 void ToolTip::buildDrawParams(std::pair<Control*, std::string>& item, int mouseX)
 {
-	const auto toolTipSize = mFont.size(item.second) + PaddingSize * 2;
+	const auto toolTipLineHeight = mFont.height();
+	const auto numberOfLines = static_cast<int>(std::count(item.second.begin(), item.second.end(), '\n') + 1);
+	const auto toolTipHeight = toolTipLineHeight * numberOfLines + PaddingSize.y * 2;
+
+	auto calculateMaxWidth = [this](const std::string& text) {
+		int maxWidth = 0;
+		processLines(text, [this, &maxWidth](const std::string& line) {
+			maxWidth = std::max(maxWidth, mFont.size(line).x);
+		});
+		return maxWidth;
+	};

```

Comment:
Might make sense to pull this section out as a standalone function. If `processLines` were standalone, perhaps in a unnamed namespace, then this could be made standalone too.